### PR TITLE
Upgrade JavaParser to 3.10.0

### PR DESCRIPTION
**Pull request by matozoid**
Upgrade to the latest version. Be aware: I don't know how to test this (but as the maintainer of JavaParser I'm 95% sure this is the right change) and this needs Java 8.

### CODE REVIEW

File: extensions/gdx-jnigen/src/com/badlogic/gdx/jnigen/parsing/RobustJavaMethodParser.java [hunk 1]
```diff
@@ -122,16 +121,16 @@ private void alignMethodBodies (ArrayList<JavaMethod> methods, ArrayList<JniSect
...
-			for (BodyDeclaration member : type.getMembers()) {
-				if (member instanceof ClassOrInterfaceDeclaration || member instanceof EnumDeclaration) {
+			for (BodyDeclaration<?> member : type.getMembers()) {
+				if (member.isClassOrInterfaceDeclaration() || member.isEnumDeclaration()) {
 					getJavaMethods(methods, (TypeDeclaration)member);
 				} else {
 					if (member instanceof MethodDeclaration) {
 						MethodDeclaration method = (MethodDeclaration)member;
-						if (!ModifierSet.hasModifier(((MethodDeclaration)member).getModifiers(), ModifierSet.NATIVE)) continue;
+						if(method.isNative()) continue;
```
Comment (MobiDevelop) [inline extensions/gdx-jnigen/src/com/badlogic/gdx/jnigen/parsing/RobustJavaMethodParser.java]: I think you're missing the `!` here.

Comment (matozoid) [reply extensions/gdx-jnigen/src/com/badlogic/gdx/jnigen/parsing/RobustJavaMethodParser.java]: Oops, you're right!

**Comment by matozoid** [comment]
Sorry for the delay, I think I missed a notification.

**Comment by PokeMMO** [comment]
It looks like you accidentally modified all of the whitespace

**Pull request closed without merging**